src/libawkward/layoutbuilder/ListArrayBuilder.cpp

```diff
@@ -66,18 +66,26 @@ namespace awkward {
     const ForthOutputBufferMap& outputs) const {
     auto search = outputs.find(vm_output_data());
     if (search != outputs.end()) {
-      auto offsets = search->second.get()->toIndex64();
+      auto output = search->second;
+      if (output->dtype() != util::dtype::int64) {
```

Comment:
I don't think this check is needed - the output type is encoded by a generated Forth code.

Reply:
Hi @ianna, thanks for reviewing! Could you elaborate on this, i.e. why you think we _know_ we have an Index64? Are you saying that the Forth code that generates these buffers is producing 64 bit integers?

If so, I believe this might still be needed, as a safety mechanism (even if it's "less safe" than the previous virtual-dispatch based approach).

Reply:
The constructor of the builder generates the Forth code, that includes the type of its output buffer.